### PR TITLE
HKG: Car Port for Genesis GV70 2022

### PR DESCRIPTION
**Pull request by sunnyhaibin**
**Checklist**
- [x] added entry to CarInfo in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [x] route with openpilot: `ca4de5b12321bd98|2022-10-18--21-15-59`
- [x] harness type: [Hyundai L](https://github.com/commaai/openpilot/wiki/Hyundai-Kia-Genesis#harness-guide)

**Physical setup**
- Top port behind comma three
  - comma Power <-> First Harness Box <-> comma three USB C/Right-Angle C <-> comma three
- Bottom port behind comma three
  - Camera Harness <-> Second Harness Box <-> USB C/C 3.1 <-> Red Panda <-> USB A/A <-> USB A to C **OTG Adapter** <-> comma three

**Requires**
- https://github.com/commaai/panda/pull/1134

Thanks to community GV70 owner zunichky.

### CODE REVIEW

File: selfdrive/car/hyundai/values.py [hunk 1]
```diff
@@ -1422,6 +1424,22 @@ class Buttons:
...
+    ],
+  },
+  CAR.KIA_SPORTAGE_HYBRID_5TH_GEN: {
+    (Ecu.fwdCamera, 0x7c4, None): [
+      b'\xf1\x00NQ5 FR_CMR AT USA LHD 1.00 1.00 99211-P1060 665',
+    ],
+    (Ecu.fwdRadar, 0x7d0, None): [
+      b'\xf1\x00NQ5__               1.01 1.03 99110-CH000         ',
+    ],
+  },
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:1442]:
```suggestion
```

Why is this added?

Comment (sunnyhaibin) [reply selfdrive/car/hyundai/values.py]: Oops, thanks for catching it!

File: selfdrive/car/hyundai/values.py [hunk 2]
```diff
@@ -1439,10 +1450,10 @@ class Buttons:
   "use_fca": {CAR.SONATA, CAR.SONATA_HYBRID, CAR.ELANTRA, CAR.ELANTRA_2021, CAR.ELANTRA_HEV_2021, CAR.KIA_STINGER, CAR.IONIQ_EV_2020, CAR.IONIQ_PHEV, CAR.KONA_EV, CAR.KIA_FORTE, CAR.KIA_NIRO_EV, CAR.PALISADE, CAR.GENESIS_G70, CAR.GENESIS_G70_2020, CAR.KONA, CAR.SANTA_FE, CAR.KIA_SELTOS, CAR.KONA_HEV, CAR.SANTA_FE_2022, CAR.KIA_K5_2021, CAR.IONIQ_HEV_2022, CAR.SANTA_FE_HEV_2022, CAR.SANTA_FE_PHEV_2022, CAR.TUCSON, CAR.KONA_EV_2022},
 }
 
-CANFD_CAR = {CAR.KIA_EV6, CAR.IONIQ_5, CAR.TUCSON_HYBRID_4TH_GEN, CAR.KIA_SPORTAGE_HYBRID_5TH_GEN, CAR.SANTA_CRUZ_1ST_GEN, CAR.KIA_SPORTAGE_5TH_GEN}
+CANFD_CAR = {CAR.KIA_EV6, CAR.IONIQ_5, CAR.TUCSON_HYBRID_4TH_GEN, CAR.KIA_SPORTAGE_HYBRID_5TH_GEN, CAR.SANTA_CRUZ_1ST_GEN, CAR.KIA_SPORTAGE_5TH_GEN, CAR.GENESIS_GV70_1ST_GEN}
 
 # The camera does SCC on these cars, rather than the radar
-CAMERA_SCC_CAR = {CAR.KONA_EV_2022, }
+CAMERA_SCC_CAR = {CAR.KONA_EV_2022, CAR.KIA_EV6, CAR.TUCSON_HYBRID_4TH_GEN, CAR.KIA_SPORTAGE_HYBRID_5TH_GEN, CAR.SANTA_CRUZ_1ST_GEN, CAR.KIA_SPORTAGE_5TH_GEN}
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:1456]: This was only for platforms which all cars were camera SCC. This makes it confusing as the EV6 is not camera SCC, unless HDA2, but that's hidden in a non-obvious if statement in CarState

File: selfdrive/car/hyundai/values.py [hunk 3]
```diff
@@ -48,6 +48,7 @@ class HyundaiFlags(IntFlag):
   CANFD_HDA2 = 1
   CANFD_ALT_BUTTONS = 2
   CANFD_ALT_GEARS = 4
+  CANFD_CAMERA_SCC = 8
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:51]: We can combine in the future, for now we check CAMERA_SCC_CAR in a few places where it's difficult to add the CANFD cars (like that expLong is available).

Comment (sshane) [inline selfdrive/car/hyundai/values.py:51]: Will think about unifying this in the future as we see more cars. Just didn't like how it wouldn't be true if CANFD cars were in that list based on HDA level, so splitting and using flags for now

**Comment by sshane** [comment]
thanks for the port!

**Comment by sshane** [comment]
@zunichky want to verify, you have the Hyundai L harness>?

**Comment by zunichky** [comment]
> @zunichky want to verify, you have the Hyundai L harness?

Yep Hyundai L
